pkg/cloud-provider/cloudapi/aws/aws_crd_helpers.go

```diff
@@ -74,9 +77,58 @@ func ec2InstanceToInternalVirtualMachineObject(instance *ec2.Instance, namespace
 	cloudID := *instance.InstanceId
 	cloudNetwork := *instance.VpcId
 
-	return utils.GenerateInternalVirtualMachineObject(cloudID, strings.ToLower(cloudName), strings.ToLower(cloudID), strings.ToLower(region),
-		namespace, strings.ToLower(cloudNetwork), cloudNetwork, runtimev1alpha1.VMState(*instance.State.Name), tags, networkInterfaces,
-		providerType, account)
+	vpcName := extractVpcName(vpcs, strings.ToLower(cloudNetwork))
+
+	vmStatus := &runtimev1alpha1.VirtualMachineStatus{
+		Provider:          runtimev1alpha1.CloudProvider(providerType),
+		Tags:              tags,
+		State:             runtimev1alpha1.VMState(*instance.State.Name),
+		NetworkInterfaces: networkInterfaces,
+		Region:            strings.ToLower(region),
+		Agented:           false,
+		CloudId:           strings.ToLower(cloudID),
+		CloudName:         strings.ToLower(cloudName),
+		CloudVpcId:        strings.ToLower(cloudNetwork),
+		CloudVpcName:      vpcName,
+	}
+
+	labelsMap := map[string]string{
+		config.LabelCloudAccountName:      account.Name,
+		config.LabelCloudAccountNamespace: account.Namespace,
+		config.LabelVpcName:               strings.ToLower(cloudNetwork),
+		config.LabelCloudVmUID:            strings.ToLower(cloudID),
+		config.LabelCloudVpcUID:           strings.ToLower(cloudNetwork),
+	}
+
+	vmCrd := &runtimev1alpha1.VirtualMachine{
+		TypeMeta: v1.TypeMeta{
+			Kind:       cloudcommon.VirtualMachineRuntimeObjectKind,
+			APIVersion: cloudcommon.RuntimeAPIVersion,
+		},
+		ObjectMeta: v1.ObjectMeta{
+			UID:       uuid.NewUUID(),
+			Name:      cloudID,
+			Namespace: namespace,
+			Labels:    labelsMap,
+		},
+		Status: *vmStatus,
+	}
+
+	return vmCrd
+}
+
+// extractVpcName function extracts tag by name "Name" and returns it as vpc name.
+func extractVpcName(vpcs map[string]*ec2.Vpc, vpcId string) string {
+	var vpcName string
+	if value, found := vpcs[vpcId]; found {
+		for _, tag := range value.Tags {
+			if *(tag.Key) == "Name" {
```

Comment:
please simulate a case when vpc name is not present..

Reply:
Vpc and VM objects when vpc name is not present:
archana@archana-virtual-machine:~/nephe/yaml$ kubectl describe vpc vpc-0882e6544e1bb0564 -n sample-ns
Name:         vpc-0882e6544e1bb0564
Namespace:    sample-ns
Labels:       nephe.io/cloud-region=us-west-1
              nephe.io/cloud-vpc-uid=vpc-0882e6544e1bb0564
              nephe.io/cpa-name=cloudprovideraccount-aws-sample
              nephe.io/cpa-namespace=sample-ns
Annotations:  <none>
API Version:  runtime.cloud.antrea.io/v1alpha1
Kind:         Vpc
Metadata:
  Creation Timestamp:  <nil>
Status:
  Cidrs:
    10.10.0.0/16
  Cloud Id:  vpc-0882e6544e1bb0564
  Managed:   true
  Provider:  AWS
  Region:    us-west-1
  Tags:
    Environment:  nephe
    Terraform:    true
Events:           <none>
archana@archana-virtual-machine:~/nephe/yaml$ kubectl describe vm i-0c04dbfe5d52ea26d -n sample-ns
Name:         i-0c04dbfe5d52ea26d
Namespace:    sample-ns
Labels:       nephe.io/cloud-vm-uid=i-0c04dbfe5d52ea26d
              nephe.io/cloud-vpc-uid=vpc-0882e6544e1bb0564
              nephe.io/cpa-name=cloudprovideraccount-aws-sample
              nephe.io/cpa-namespace=sample-ns
              nephe.io/vpc-name=vpc-0882e6544e1bb0564
Annotations:  <none>
API Version:  runtime.cloud.antrea.io/v1alpha1
Kind:         VirtualMachine
Metadata:
  Creation Timestamp:  <nil>
  UID:                 bf25516e-ce92-4f89-a2ab-1cc946f88ce1
Status:
  Agented:       false
  Cloud Id:      i-0c04dbfe5d52ea26d
  Cloud Vpc Id:  vpc-0882e6544e1bb0564
  Network Interfaces:
    Ips:
      Address:       10.10.6.63
      Address Type:  InternalIP
      Address:       52.53.251.33
      Address Type:  ExternalIP
    Mac:             06:2c:53:9b:2c:fb
    Name:            eni-06ae34d1151e1cd41
  Provider:          AWS
  Region:            us-west-1
  State:             running
Events:              <none>